### PR TITLE
Implementing prettier formatting through external file

### PR DESCRIPTION
**Pull request by Ririio**
## This a new pull request based on #47 ##

Adding eslint recommendation | Creating external eslintignore file

Initial .vscode settings configuration

ignoring my-sql directory for linter

removing cypress and tailwind from prettierignore

adding telescope configuration for vscode

Adding newline at the end of each file

### CODE REVIEW

File: .eslintignore
Comment (humphd) [inline .eslintignore:4]: `/mysql-data`

File: .prettierignore
Comment (humphd) [inline .prettierignore:11]: You can remove this.

File: .prettierignore
Comment (humphd) [inline .prettierignore:9]: You can remove this too.

File: .prettierrc
Comment (humphd) [inline .prettierrc:15]: You should always include a newline at the end of your files (GitHub is marking it to show you it's missing).  The reason we do this is so that files don't get concatenated together, and the end of one file be on the same line as the start of another.

File: .vscode/settings.json
Comment (humphd) [inline .vscode/settings.json:1]: While you're in this file, can you add what we have in Telescope? https://github.com/Seneca-CDOT/telescope/blob/master/.vscode/settings.json

**Review by humphd — CHANGES_REQUESTED** [review]

File: .eslintignore
Comment (humphd) [inline .eslintignore:4]: Remove `/postgres-data`, we don't have it anymore.

File: .prettierignore
Comment (humphd) [inline .prettierignore:9]: Add `/mysql-data` so it doesn't run prettier in there (that's the new database volume).

**Review by humphd — CHANGES_REQUESTED** [review]

File: .eslintrc.js
Comment (Myrfion) [inline .eslintrc.js:11]: Since we are removing `cypress` maybe it makes sense to remove it from this file?

File: .eslintrc.js
Comment (Myrfion) [inline .eslintrc.js:13]: same thing regarding the `cypress` is there a good reason to have it here?

Comment (Ririio) [reply .eslintrc.js:13]: I wasn't sure if I should remove it considering someone was already given a task regarding removal of cypress and tailwind from our repo

File: .prettierrc
Comment (dadolhay) [inline .prettierrc:5]: A lot of these are already the default, like
"tabWidth": 2,
"trailingComma": "es5",
"jsxSingleQuote": false,
 "bracketSpacing": true,
 etc.
 
 I would not hardwire in the defaults again

File: .eslintrc.js
Comment (dadolhay) [reply .eslintrc.js:13]: I would suggest to remove those too

File: .eslintrc.js
Comment (dadolhay) [reply .eslintrc.js:11]: I would suggest to remove those too

**Review by dadolhay — CHANGES_REQUESTED** [review]

File: .prettierrc
Comment (Ririio) [reply .prettierrc:5]: What would happen if a contributor has their `tabwidth` set to a different value? I thought we implement them here to keep consistency just in case some people have their settings altered?

Comment (humphd) [reply .prettierrc:5]: One of the reasons people specify the defaults for Prettier like this is that the project has been known to change the defaults in the past.  What then happens is you update the prettier dependency, and your entire tree gets reformatted for no good reason.

I think explicit defaults are useful, even if only to document our decisions.

**Review by humphd — DISMISSED** [review]

**Review by humphd — APPROVED** [review]

**Comment by shawnyu5** [comment]
This branch seems to have conflicts with main...

**Comment by humphd** [comment]
cc @Eakam1007, who just wrote this stuff for Chakra UI

**Comment by humphd** [comment]
you still have conflicts here.

**Comment by Ririio** [comment]
> you still have conflicts here.

It was conflict through imports, it should be fixed now

**Comment by humphd** [comment]
You are failing prettier CI now.  Please get this passing CI and mergable.

**Comment by shawnyu5** [comment]
ahh, since this PR changes `.prettierrc`, prettier on this branch will format according to this PRs prettierrc, which is different than the one on master. Hence prettier check will always fail

**Comment by shawnyu5** [comment]
I think we can just merge with the failing prettier check for now. When this PR lands, CI on main will use the new prettierrc, and will pass CI again.

Tho this PR handling the reformatting of all the files is a little risky. Ideally, formatting all files should've been its own separate PR. But before we merge this, we got to make sure we only changed formatting from main, not any of the code contents

**Comment by Ririio** [comment]
> I think we can just merge with the failing prettier check for now. When this PR lands, CI on main will use the new prettierrc, and will pass CI again.
> 
> Tho this PR handling the reformatting of all the files is a little risky. Ideally, formatting all files should've been its own separate PR. But before we merge this, we got to make sure we only changed formatting from main, not any of the code contents

If we're not going to worry about the prettier, I can roll it back to the commit that didn't affect the files?

**Comment by shawnyu5** [comment]
If we don't worry about formatting in this PR, when we merge, prettier check on main will fail. A separate PR would be needed to reformat all files. This is a little safer and keeps the scope of the PR limited

**Comment by Ririio** [comment]
> If we don't worry about formatting in this PR, when we merge, prettier check on main will fail. A separate PR would be needed to reformat all files. This is a little safer and keeps the scope of the PR limited

I rolled it back to the original commit without any formatting. There shouldn't be any issue when merging now

**Comment by shawnyu5** [comment]
This PR adds a `.prettierrc`, and reformats all files to use the new formatting standard

File: .prettierrc
Comment (humphd) [inline .prettierrc:2]: This should be 2

**Review by humphd — CHANGES_REQUESTED** [review]

File: .prettierrc
Comment (shawnyu5) [reply .prettierrc]: oops, forgot to change it back after testing

**Comment by Ririio** [comment]
There's always bound to be a conflict, and prettier will always fail unless we push this pr. We can roll it back to the commit which only affects 5 files as to the entire repo...

**Comment by shawnyu5** [comment]
Prettier should not always fail, [this commit](https://github.com/Seneca-CDOT/starchart/pull/56/commits/9e21218c3a3ddc832eba8575da63b65c5565e71c) has prettier passing ci

**Comment by Ririio** [comment]
> Prettier should not always fail, [this commit](https://github.com/Seneca-CDOT/starchart/pull/56/commits/9e21218c3a3ddc832eba8575da63b65c5565e71c) has prettier passing ci

I'm referring to it consistently failing whenever a new a commit is pass. We can fix those issues sure, but isn't it better to send the settings now and do formatting later?

File: .eslintignore
Comment (dadolhay) [inline .eslintignore:4]: Do we use postgres?

File: .eslintrc.js
Comment (dadolhay) [reply .eslintrc.js:10]: I would remove this because if the cypress replace PR will have to, it will cause a conflict. Easier to do here and now

File: .eslintrc.js
Comment (dadolhay) [reply .eslintrc.js:12]: same as above

File: .prettierignore
Comment (dadolhay) [reply .prettierignore:9]: ^^

File: .prettierignore
Comment (dadolhay) [reply .prettierignore:9]: ^^

File: .eslintignore
Comment (dadolhay) [inline .eslintignore:5]: /mysql-data does not contain js/ts code, no need to add it

**Review by dadolhay — CHANGES_REQUESTED** [review]

**Review by dadolhay — APPROVED** [review]
This will cause some conflicts for sure, but at least the code becomes consistent with the new rules

File: .eslintignore
Comment (shawnyu5) [inline .eslintignore]: Why do we want to lint the sql data folder?

I get weird errors if the sql data folder is linted

image

Comment (shawnyu5) [reply .eslintignore]: Same issue with `npm run format`

Comment (dadolhay) [reply .eslintignore]: Only js, ts jsx and tsx files should be ever linted. Give me a sec

**Review by dadolhay — APPROVED** [review]

File: .eslintignore
Comment (shawnyu5) [reply .eslintignore]: We have other file types in the repo, like markdown and yaml. I think we should be excluding the stuff we dont want to lint / format, rather than including everything we would like to

Comment (shawnyu5) [reply .eslintignore]: theres also a `.js.map`

**Comment by shawnyu5** [comment]
Still the same error
image

File: .eslintignore
Comment (dadolhay) [reply .eslintignore]: The issue is, that that folder it references was created by docker, and it cannot scan the folder for js/ts/jsx/tsx files so it fails.
You are right, because of this mysql-data has to be added to eslintignore and prettierignore

**Review by dadolhay — DISMISSED** [review]

**Comment by Ririio** [comment]
@shawnyu5 do you mind approving it if there's no more mistakes?

**Comment by shawnyu5** [comment]
A hack we can do, what I've been doing. For the format script, we can change it to `prettier --write * .*`. That seems to carry on despite the error.

**Comment by Ririio** [comment]
> A hack we can do, what I've been doing. For the format script, we can change it to `prettier --write * .*`. That seems to carry on despite the error.

what does it do?

**Comment by Ririio** [comment]
> A hack we can do, what I've been doing. For the format script, we can change it to `prettier --write * .*`. That seems to carry on despite the error.

what does it do?

**Comment by dadolhay** [comment]
https://github.com/prettier/prettier/issues/11568

**Review by dadolhay — APPROVED** [review]
Since this is not a new issue, I think the prettier problem should be handled in a different ticket. I think this is good to merge

**Comment by shawnyu5** [comment]
I dont think we want to merge with a broken prettier script. Other people will have a pretty big headache formatting their code....

**Comment by humphd** [comment]
Closing in favour of #94

**Pull request closed without merging**
